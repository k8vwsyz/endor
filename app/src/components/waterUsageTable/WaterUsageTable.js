import React, { useCallback, useEffect, useState } from "react";
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  TableSortLabel,
  Toolbar,
  Divider,
  TextField,
  IconButton
} from '@material-ui/core'
import Paper from '@material-ui/core/Paper'
import FilterListIcon from '@material-ui/icons/FilterList'
import { makeStyles } from "@material-ui/core/styles";

import { sortEntries, getComparator } from './helperModules/SortEntries'
import { filterCountries } from './helperModules/FilterList'
import { debounce } from 'throttle-debounce'

const useStyles = makeStyles(() => ({
  waterUsageToolbar: {
    justifyContent: "space-between",
  }
}));

const headCells = [
  { id: "country", numeric: false, label: "Country" },
  { id: "year", numeric: true, label: "Year" },
  { id: "volume", numeric: true, label: "Usage (Billion mÂ³)" }
]

// ---------------------------------
// Table head
// ---------------------------------
const WaterUsageTableHead = props => {
  const { orderBy, order, onRequestSort } = props

  const createSortHandler = (property) => (event) => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        {headCells.map((cell) => {
          return (
            <TableCell
              key={cell.id}
              align={cell.numeric ? "right" : "left"}>
              <TableSortLabel
                active={orderBy === cell.id}
                direction={orderBy === cell.id ? order : 'asc'}
                onClick={createSortHandler(cell.id)}>
                {cell.label}
              </TableSortLabel>
            </TableCell>
          )
        })
        }
      </TableRow>
    </TableHead>
  )
}

// ---------------------------------
// Table body
// ---------------------------------
const WaterUsageTableBody = props => {
  const { entries } = props

  return (
    <TableBody data-testid="water-usage">
      {entries.map((row, key) => (
        <TableRow key={key}>
          <TableCell component="th" scope="row">{row.country}</TableCell>
          <TableCell align="right">{row.year}</TableCell>
          <TableCell align="right">{row.volume}</TableCell>
        </TableRow>
      ))}
    </TableBody>
  )
}

// ---------------------------------
// Table toolbar
// ---------------------------------
const WaterUsageToolbar = props => {
  const classes = useStyles()
  const { onFilterRequested } = props
  const [search, setSearch] = useState("")
  const debouncedSearch = useCallback(debounce(300, false, (s) => onFilterRequested(s)), [])

  useEffect(() => () => debouncedSearch.cancel(), [debouncedSearch])

  const handleChange = event => {
    const filterText = event.target.value
    setSearch(filterText)
    debouncedSearch(filterText)
  }

  const handleEnter = event => {
    if (event.charCode === 13)
      onFilterRequested(search)
  }

  const filterRequested = _event => {
    onFilterRequested(search)
  }

  return (
    <Toolbar className={classes.waterUsageToolbar}>
      <TextField
        data-testid="filter-input"
        variant="filled"
        label="Filter"
        placeholder="Country name..."
        onChange={handleChange}
        onKeyPress={handleEnter}
        value={search}
      />

      <IconButton data-testid="filter-button" onClick={filterRequested}>
        <FilterListIcon />
      </IconButton>
    </Toolbar>
  )
}

// ---------------------------------
// Table
// ---------------------------------
const WaterUsageTable = props => {
  const { data } = props

  const [entries, setEntries] = useState(data)
  const [order, setOrder] = useState('asc')
  const [orderBy, setOrderBy] = useState('country')

  const handleRequestSort = (_event, property) => {
    const isAsc = orderBy === property && order === 'asc'
    const _order = isAsc ? 'desc' : 'asc'
    const _orderBy = property

    const comparator = getComparator(_order, _orderBy)
    const sortedEntries = sortEntries(entries, comparator)

    setOrder(_order)
    setOrderBy(_orderBy)
    setEntries(sortedEntries)
  }

  const handleRequestFilter = (filterString) => {
    const filteredCountries = filterCountries(filterString, data)
    setEntries(filteredCountries)
  }

  return (
    <Paper>
      <WaterUsageToolbar onFilterRequested={handleRequestFilter} />
      <Divider />
      <TableContainer>
        <Table>
          <WaterUsageTableHead
            orderBy={orderBy}
            order={order}
            onRequestSort={handleRequestSort} />
          <WaterUsageTableBody entries={entries} />
        </Table>
      </TableContainer>
    </Paper>
  );
}

export default WaterUsageTable