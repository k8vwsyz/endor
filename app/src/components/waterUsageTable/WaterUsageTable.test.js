import React from "react";
import { render, cleanup, act } from "@testing-library/react";
import { screen, waitFor } from "@testing-library/dom";
import WaterUsageTable from './WaterUsageTable'
import userEvent from '@testing-library/user-event'

afterEach(cleanup);

const data = [
  {
    country: "Albania",
    year: 1990,
    volume: 1.2,
  },
  {
    country: "Azerbaijan",
    year: 2012,
    volume: 11.97,
  },
]

it("snapshot test", async () => {
  const { getByTestId, asFragment } = render(<WaterUsageTable data={data} />)

  await act(async () => waitFor(() => expect(getByTestId("water-usage").childElementCount).toBe(2)))

  expect(asFragment()).toMatchSnapshot()
})

describe("filter tests", () => {
  it("filter 'a'", async () => {
    const { asFragment } = render(<WaterUsageTable data={data} />)

    await act(async () => waitFor(() => expect(screen.getByTestId("water-usage").childElementCount).toBe(2)))

    userEvent.type(screen.getByTestId("filter-input").children[1].children[0], "a")
    userEvent.click(screen.getByTestId("filter-button"))

    expect(screen.getByTestId("water-usage").childElementCount).toBe(2)
    expect(asFragment()).toMatchSnapshot()
  })

  it("filter 'alb'", async () => {
    const { asFragment } = render(<WaterUsageTable data={data} />)

    await act(async () => waitFor(() => expect(screen.getByTestId("water-usage").childElementCount).toBe(2)))

    userEvent.type(screen.getByTestId("filter-input").children[1].children[0], "alb")
    userEvent.click(screen.getByTestId("filter-button"))

    expect(screen.getByTestId("water-usage").childElementCount).toBe(1)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("order tests", () => {
  it("order by country descending", async () => {
    const { asFragment } = render(<WaterUsageTable data={data} />)

    await act(async () => waitFor(() => expect(screen.getByTestId("water-usage").childElementCount).toBe(2)))

    userEvent.click(screen.getByRole("button", { name: "Country" }))

    expect(asFragment()).toMatchSnapshot()
  })

  it("order by year ascending", async () => {
    const { asFragment } = render(<WaterUsageTable data={data} />)

    await act(async () => waitFor(() => expect(screen.getByTestId("water-usage").childElementCount).toBe(2)))

    userEvent.click(screen.getByTestId("filter-button"))

    userEvent.click(screen.getByRole("button", { name: "Year" }))

    expect(asFragment()).toMatchSnapshot()
  })
})