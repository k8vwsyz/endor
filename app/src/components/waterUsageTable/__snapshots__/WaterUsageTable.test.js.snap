// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter tests filter 'a' 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-waterUsageToolbar-2 MuiToolbar-gutters"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        data-testid="filter-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
          data-shrink="true"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            placeholder="Country name..."
            type="text"
            value="a"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="filter-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        >
          <span
            class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
            style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
          >
            <span
              class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
            />
          </span>
        </span>
      </button>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                role="button"
                tabindex="0"
              >
                Country
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Year
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Usage (Billion mÂ³)
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
          data-testid="water-usage"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Albania
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1990
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.2
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Azerbaijan
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2012
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              11.97
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`filter tests filter 'alb' 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-waterUsageToolbar-3 MuiToolbar-gutters"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        data-testid="filter-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
          data-shrink="true"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            placeholder="Country name..."
            type="text"
            value="alb"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="filter-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        >
          <span
            class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
            style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
          >
            <span
              class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
            />
          </span>
        </span>
      </button>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                role="button"
                tabindex="0"
              >
                Country
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Year
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Usage (Billion mÂ³)
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
          data-testid="water-usage"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Albania
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1990
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`order tests order by country descending 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-waterUsageToolbar-4 MuiToolbar-gutters"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        data-testid="filter-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            placeholder="Country name..."
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="filter-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                role="button"
                tabindex="0"
              >
                Country
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Year
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Usage (Billion mÂ³)
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
          data-testid="water-usage"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Azerbaijan
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2012
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              11.97
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Albania
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1990
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`order tests order by year ascending 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-waterUsageToolbar-5 MuiToolbar-gutters"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        data-testid="filter-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            placeholder="Country name..."
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="filter-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        >
          <span
            class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
            style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
          >
            <span
              class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
            />
          </span>
        </span>
      </button>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Country
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                role="button"
                tabindex="0"
              >
                Year
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Usage (Billion mÂ³)
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
          data-testid="water-usage"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Albania
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1990
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.2
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Azerbaijan
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2012
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              11.97
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-waterUsageToolbar-1 MuiToolbar-gutters"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        data-testid="filter-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            placeholder="Country name..."
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="filter-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                role="button"
                tabindex="0"
              >
                Country
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Year
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Usage (Billion mÂ³)
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
          data-testid="water-usage"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Albania
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1990
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.2
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Azerbaijan
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2012
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              11.97
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
